src/App.tsx

```diff
@@ -2,43 +2,81 @@ import { ChangeEvent, useState } from 'react';
 import NewTask from './components/NewTask';
-
-
+import 'semantic-ui-css/semantic.min.css';
 
 function App() {
-    const [tasksArray, setTasksArray] = useState([{ id: 0, item: "Buy Milk" }, { id: 1, item: "Go Run" }, { id: 2, item: "Make Dinner" }, { id: 3, item: "Bake Cake" }]);
-    const [newTaskItem, setNewTaskItem] = useState('')
+    const [tasksArray, setTasksArray] = useState([
+        { id: 0, item: 'Buy Milk' },
+        { id: 1, item: 'Go Run' },
+        { id: 2, item: 'Make Dinner' },
+        { id: 3, item: 'Bake Cake' },
+    ]);
+    const [newTaskItem, setNewTaskItem] = useState('');
+    const [editedTaskItem, setEditedTaskItem] = useState('');
+    const [editTaskId, setEditTaskId] = useState<number | undefined>();
 
     function textTyping(event: ChangeEvent<HTMLInputElement>) {
-        setNewTaskItem(event.target.value)
+        const { name } = event.target;
+        name === 'new-task-text'
+            ? setNewTaskItem(event.target.value)
+            : setEditedTaskItem(event.target.value);
     }
 
-    function clearTyping(){
-        setNewTaskItem("")
+    function clearTyping() {
+        setNewTaskItem('');
     }
 
     function addItem() {
-        setTasksArray([...tasksArray, { id: (tasksArray.length > 0 ) ? tasksArray.slice(-1)[0].id + 1 : 0, item: newTaskItem }])
-        setNewTaskItem("")
-     }
+        setTasksArray([
+            ...tasksArray,
+            { id: tasksArray.length > 0 ? tasksArray.slice(-1)[0].id + 1 : 0, item: newTaskItem },
+        ]);
+        setNewTaskItem('');
+    }
 
     function deleteTask(id: number) {
-        const newTasksArray = tasksArray.filter((item) => item.id !== id)
-        setTasksArray(newTasksArray)
+        const newTasksArray = tasksArray.filter((item) => item.id !== id);
+        setTasksArray(newTasksArray);
     }
 
     function clearTasksList() {
-        setTasksArray([])
+        setTasksArray([]);
+        setEditTaskId(undefined);
+    }
+
+    function editTask(id: number) {
+        setEditTaskId(id);
+    }
+
+    function saveItem(idToSave: number, itemToSave: string) {
+        tasksArray.splice(idToSave, 1, { id: idToSave, item: itemToSave });
+
+        setTasksArray([...tasksArray]); // clone
+        setEditTaskId(undefined);
+        setEditedTaskItem('');
     }
 
     return (
         <main className={styles.main}>
             <h2 className={styles.title}>My ToDo List</h2>
-            <TasksList items={tasksArray} onDelete={deleteTask} clearAll={clearTasksList} />
-            <NewTask item={newTaskItem} typeText={textTyping} onAdd={addItem} onClear={clearTyping} />
+            <TasksList
+                item={editedTaskItem}
+                itemsArr={tasksArray}
+                editedTaskId={editTaskId}
+                onDelete={deleteTask}
+                onEdit={editTask}
+                typeTextToEdit={textTyping}
+                onSave={saveItem}
+                clearAll={clearTasksList}
```

Comment:
Since these props are not generic, some of the variables you pass them can have the same name or something similar.